rxandroidble/src/main/java/com/polidea/rxandroidble/exceptions/BleIllegalOperationException.java

```diff
@@ -1,11 +1,24 @@
 package com.polidea.rxandroidble.exceptions;
 
+import java.util.UUID;
+
 /**
- *
+ * This exception is thrown when a non-supported operation has been requested upon a characteristic, eg. write operation on a
+ * characteristic with only read property.
  */
 public class BleIllegalOperationException extends RuntimeException {
 
-    public BleIllegalOperationException(String message) {
+    public final UUID characteristicUUID;
+    public final int supportedProperties;
+    public final int neededProperties;
+
+    public BleIllegalOperationException(String message,
```

Comment:
Well, the first thought is "Why do I need an int?! It's ugly.". If we'd have an API to read properties more easily.